hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/QuotaUtil.java

```diff
@@ -55,8 +60,36 @@ public static long getReplicatedSize(
               + partialFirstChunk * rc.getParity();
       return dataSize + replicationOverhead;
     } else {
+      LOG.warn("Unknown replication type '{}'. Returning original data size.",
+          repConfig.getReplicationType());
       return dataSize;
     }
   }
 
+  /**
+   * Get an estimated data size (before replication) from the replicated size.
+   * An (inaccurate) reverse of getReplicatedSize().
+   * @param replicatedSize size after replication.
+   * @param repConfig The replicationConfig used to store the data.
+   * @return Data size before replication.
+   */
+  public static long getDataSize(long replicatedSize,
+                                 ReplicationConfig repConfig) {
+    if (repConfig.getReplicationType() == RATIS) {
+      final int ratisReplicationFactor = ((RatisReplicationConfig) repConfig)
+          .getReplicationFactor().getNumber();
+      // May not be divisible. But it's fine to ignore remainder in our use case
+      return replicatedSize / ratisReplicationFactor;
+    } else if (repConfig.getReplicationType() == EC) {
+      ECReplicationConfig rc = (ECReplicationConfig) repConfig;
+      // In the case of EC, we won't know if keys have partial chunks or not,
+      // so we assume no partial chunks as an estimate.
+      return replicatedSize * rc.getData() / rc.getRequiredNodes();
```

Comment:
For a rough estimate, This should be fine because only the last strip will be partial.